OmfIngressClientLibraries/Program.cs

```diff
@@ -134,7 +156,38 @@ public static async Task DeleteTypeAndContainerAsync()
         public static async Task DeleteOmfConnectionAsync(OmfConnection omfConnection)
         {
             // Delete the Connection           
-            await _omfIngressClient.DeleteOmfConnectionAsync(omfConnection).ConfigureAwait(false);
+            if (omfConnection == null)
+            {
+                throw new ArgumentException("Omf Connection cannot be null", nameof(omfConnection));
+            }
+
+            // Delete the Connection
+            Console.WriteLine($"Deleting the OMF Connection with Id {omfConnection.Id}");
+            Console.WriteLine();
+
+            await _omfIngressService.BeginDeleteOmfConnectionAsync(omfConnection.Id).ConfigureAwait(false);
+
+            bool deleted = false;
+            while (!deleted)
```

Comment:
same comment as above, comment and failsafe

Reply:
Added cancellation token and throw error on a timeout